include/ideep/operators/conv.hpp

```diff
@@ -1395,10 +1396,11 @@ struct convolution_forward
         padding_l,
         padding_r,
         attr,
-        omp_get_max_threads());
-    return fetch_or_create(key, [&]() {
-      if (with_bias) {
-        return primitive_desc(
+        omp_get_max_threads(),
+        weights_hashkey);
+    dnnl::convolution_forward::primitive_desc pd;
+    if (with_bias) {
+      pd = primitive_desc(
```

Comment:
Primitive descriptor creation takes time so it is not a good idea to create it every time. That's why we cache the pd.